stage #3/java/module #2. REST API Basics/task02/repository/src/main/java/by/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,105 @@
+package by.epam.esm.dao.impl;
+
+import by.epam.esm.dao.CrdTagDao;
+import by.epam.esm.dao.mapper.TagRowMapper;
+import by.epam.esm.enity.Pagination;
+import by.epam.esm.enity.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class TagDaoImpl implements CrdTagDao {
+    private static final String FIND_ALL_REQUEST = "SELECT tag_id, name FROM tag WHERE remove=FALSE LIMIT ?, ?";
```

Comment:
Not sure these sql queries should be here, you can create separate class for them